### PR TITLE
New Basemaps: Editing Category

### PR DESCRIPTION
**Pull request by jenmerritt**
### Description

This PR is for the following sample categories:
- Editing

### General comments on Basemap updates:

It updates `Map` and/or `Basemap` Constructors as per the new design. This may include the following changes depending on the sample:
- Replace `Basemap.Type` with `BasemapStyle`
- If a lat/long/zoom was being set in the `Map` constructor, this is replaced with a `Viewpoint` being set on the `MapView`.
- Any Readme Updates identified

In addition, the samples that include the above changes, or other services requiring an API Key going forward, have the API Key  set within the sample. This will then work in conjunction with the Gradle Task being added in PR 593.

Once 593 is approved and merged into `dev` we will merge those changes into these PRs and then update all the Samples to `100.10.0-xxxx` and do a round of testing of the Samples with all the API Key / Basemap changes at that time. (We have left them as 100.9.0 (in most cases) for now to minimize merge conflicts in the `build.gradle` file).

In the meantime, the updates listed above can be reviewed. Thanks!

### CODE REVIEW

File: editing/add-features/src/main/java/com/esri/samples/add_features/AddFeaturesSample.java [hunk 1]
```diff
@@ -66,11 +68,19 @@ public void start(Stage stage) {
...
       stage.show();
 
-      // create a map with streets basemap
-      ArcGISMap map = new ArcGISMap(Basemap.Type.STREETS, 40, -95, 4);
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
 
-      // create a view for this ArcGISMap
+      // create a map with streets basemap style
```
Comment (JonLavi) [inline editing/add-features/src/main/java/com/esri/samples/add_features/AddFeaturesSample.java]: Maybe a 'the' will make this more plain english?
```suggestion
      // create a map with the streets basemap style
```

File: editing/delete-features/src/main/java/com/esri/samples/delete_features/DeleteFeaturesSample.java [hunk 2]
```diff
@@ -90,11 +96,15 @@ public void start(Stage stage) {
         });
       });
 
-      // create a map with streets basemap
-      ArcGISMap map = new ArcGISMap(Basemap.Type.STREETS, 40, -95, 4);
+      // create a map with streets basemap style
```
Comment (JonLavi) [inline editing/delete-features/src/main/java/com/esri/samples/delete_features/DeleteFeaturesSample.java]: Same here?
```suggestion
      // create a map with the streets basemap style
```

File: editing/edit-feature-attachments/src/main/java/com/esri/samples/edit_feature_attachments/EditFeatureAttachmentsSample.java [hunk 3]
```diff
@@ -113,13 +119,16 @@ public void start(Stage stage) {
       // add controls to the panel
       controlsVBox.getChildren().addAll(addAttachmentButton, deleteAttachmentButton, attachmentsLabel, attachmentList);
 
-      // create a map view
-      mapView = new MapView();
+      // create a map with streets basemap style
```
Comment (JonLavi) [inline editing/edit-feature-attachments/src/main/java/com/esri/samples/edit_feature_attachments/EditFeatureAttachmentsSample.java]: Another one? (I won't comment on the rest)
```suggestion
      // create a map with the streets basemap style
```

File: editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java [hunk 4]
```diff
@@ -69,13 +71,20 @@ public void start(Stage stage) {
       stage.setScene(scene);
       stage.show();
 
-      // create the map with a light gray canvas basemap centered on Loudoun, Virginia
-      ArcGISMap map = new ArcGISMap(Basemap.Type.LIGHT_GRAY_CANVAS_VECTOR, 39.0204, -77.4159, 18);
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create the map with a light gray basemap style
```
Comment (JonLavi) [inline editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java]: The style?
```suggestion
      // create the map with the light gray basemap style
```

**Review by JonLavi — APPROVED** [review]
lgtm!

File: editing/edit-feature-attachments/src/main/java/com/esri/samples/edit_feature_attachments/EditFeatureAttachmentsSample.java (hunk 3 above)
Comment (jenmerritt) [reply editing/edit-feature-attachments/src/main/java/com/esri/samples/edit_feature_attachments/EditFeatureAttachmentsSample.java]: Have been through all changed samples and made this tidy up

File: editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java [hunk 5]
```diff
@@ -69,13 +71,20 @@ public void start(Stage stage) {
       stage.setScene(scene);
       stage.show();
 
-      // create the map with a light gray canvas basemap centered on Loudoun, Virginia
-      ArcGISMap map = new ArcGISMap(Basemap.Type.LIGHT_GRAY_CANVAS_VECTOR, 39.0204, -77.4159, 18);
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a map with a basemap style
```
Comment (Rachael-E) [inline editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java]: to be consistent with the comment style used for topographic and streets basemap, we could use the following comment unless there was a reason to keep it generic? this change would apply across a number of these classes if you want to go for it. 

```suggestion
      // create a map with the light gray basemap style
```

**Review by Rachael-E — APPROVED** [review]
@jenmerritt looks good: I've queried the consistency of the basemap comment across classes. if you think it's too much work to go through for these PRs we can do it another time, it's not urgent. Approving so you can merge if it looks too much 👍

File: editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java (hunk 5 above)
Comment (jenmerritt) [reply editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationSample.java]: For reference, we are going with `create a map with the light gray basemap style` for whatever basemap style is being set.

**Comment by jenmerritt** [comment]
`dev` merged in (including API Key gradle tasks changes). Samples that have an API Key set have been tested running from the IDE.

Await merging these changes until release.

**Pull request merged**